src/seedu/addressbook/AddressBook.java

```diff
@@ -987,9 +987,12 @@ private static String encodePersonToString(HashMap<PersonProperty, String> perso
 	private static boolean isPersonDataExtractableFrom(String personData) {
 		final String matchAnyPersonDataPrefix = PERSON_DATA_PREFIX_PHONE + '|' + PERSON_DATA_PREFIX_EMAIL;
 		final String[] splitArgs = personData.trim().split(matchAnyPersonDataPrefix);
-		return splitArgs.length == 3 // 3 arguments
+		
+		boolean hasThreeArgs = splitArgs.length == 3;
```

Comment:
oh! you have done something here that is meaningful, but still why `hasThreeArgs` why not 4 or 5 ...